### PR TITLE
Replace compress and decompress program

### LINKED ISSUE #658: Replace compress_program and decompress_program
### Feature Request

It should be done after #649 is merged

### PR DESCRIPTION
**Pull request by cptartur**
Closes #658 

## Introduced changes

- Replaces compress and decompress program from cairo lang

##

- [x] This PR contains breaking changes

- Decompress program failure will no longer raise a StarkException

### CODE REVIEW

File: starknet_py/net/models/transaction.py [hunk 1]
```diff
@@ -105,6 +120,20 @@ def calculate_hash(self, chain_id: StarknetChainId) -> int:
             nonce=self.nonce,
         )
 
+    @staticmethod
+    def _contract_attribute_name(data: Dict[str, Any]):
+        return (
+            "contract_definition" if "contract_definition" in data else "contract_class"
```
Comment (drknzz) [inline starknet_py/net/models/transaction.py:126]: I think it's backward compatibility, no?

Comment (cptartur) [reply starknet_py/net/models/transaction.py:126]: It is. It's not expensive to have it though, so I think it's fine.

**Review by drknzz — APPROVED** [review]

File: starknet_py/net/models/transaction.py [hunk 2]
```diff
@@ -105,6 +120,20 @@ def calculate_hash(self, chain_id: StarknetChainId) -> int:
...
+        )
+
+    @staticmethod
+    def _compress_program(program: Dict[str, Any]):
+        compressed_program = json.dumps(program)
+        compressed_program = gzip.compress(data=compressed_program.encode("ascii"))
+        compressed_program = base64.b64encode(compressed_program)
+
+        return compressed_program.decode("ascii")
```
Comment (war-in) [inline starknet_py/net/models/transaction.py:135]: Do we really need these two as static methods?

Comment (war-in) [reply starknet_py/net/models/transaction.py:135]: Ok, I see what is the problem. Let's keep it as it is

**Review by war-in — APPROVED** [review]

File: starknet_py/net/models/transaction.py (hunk 2 above)
Comment (cptartur) [reply starknet_py/net/models/transaction.py:135]: Compress program could be inlined. Attribute name would be duplicated in 2 places.

File: starknet_py/net/models/transaction.py [hunk 3]
```diff
@@ -105,6 +124,12 @@ def calculate_hash(self, chain_id: StarknetChainId) -> int:
             nonce=self.nonce,
         )
 
+    @staticmethod
+    def _contract_attribute_name(data: Dict[str, Any]):
+        return (
+            "contract_definition" if "contract_definition" in data else "contract_class"
+        )
```
Comment (cptartur) [inline starknet_py/net/models/transaction.py:131]: I wanted to test serializing contract that has `contract_definition` defined, but I went through a couple of old compiled contract I had and none of them had this. It must have been changed a while ago or I am looking in the wrong place.

**Review by drknzz — APPROVED** [review]

**Pull request merged**